src/v/cluster_link/replication/partition_replicator.cc

```diff
@@ -199,6 +199,7 @@ ss::future<> partition_replicator::replicate_and_wait(
             .finally(
               [inflight = std::move(inflight), data = std::move(data)]() {});
       });
+    co_await _sink->maybe_sync_pid();
```

Comment:
regarding my other comment about making this part of implementation, can we call this from within local_partition_sink::replicate() method instead of here.